### PR TITLE
[BE] #Task 210596 Get user cohort new flow

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/210596

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **Refactor**
  - Enhanced the data retrieval logic in user services to fetch a broader range of information and improve result grouping.

### CODE REVIEW

File: src/src/user/user.service.ts [hunk 1]
```diff
@@ -1472,12 +1472,16 @@ export class UserService {
...
+		sql = `SELECT ay.name as academic_year_name, ay.id as academic_year_id,p.name,p.id as program_id
+		FROM ${primary_table} pu
+		LEFT JOIN programs p  ON pu.program_id = p.id
+		LEFT JOIN academic_years ay ON pu.academic_year_id = ay.id
+		LEFT JOIN program_organisation po ON pu.program_id = po.program_id
+		WHERE po.status = 'active' AND po.organisation_id IN (${parent_ip_id}) AND pu.user_id = ${user_id}
+		GROUP BY ay.id ,p.id
+		
+		
+		`;
```
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/user/user.service.ts:1484]: ORM is  not used .

File: src/src/facilitator/facilitator.service.ts [hunk 2]
```diff
@@ -7,8 +7,8 @@ import { UploadFileService } from 'src/upload-file/upload-file.service';
 import { UserService } from 'src/user/user.service';
 import { EnumService } from '../enum/enum.service';
 import {
-    HasuraService,
-    HasuraService as HasuraServiceFromServices
+	HasuraService,
+	HasuraService as HasuraServiceFromServices,
```
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/facilitator/facilitator.service.ts:11]: both are needed

File: src/src/user/user.service.ts [hunk 3]
```diff
@@ -1500,4 +1504,68 @@ export class UserService {
...
+				success: true,
+				data: academic_data,
+			});
+		} else {
+			return res.json({
+				status: 500,
+				data: {},
+				success: false,
+			});
+		}
```
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/user/user.service.ts:1537]: grapql throws error with this.

File: src/src/user/user.service.ts [hunk 4]
```diff
@@ -1500,4 +1504,68 @@ export class UserService {
...
+				success: true,
+				data: program_data,
+			});
+		} else {
+			return res.json({
+				status: 500,
+				data: {},
+				success: false,
+			});
+		}
```
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/user/user.service.ts:1569]: grapql throws error with this.

**Comment by manojLondhe** [comment]
closing in favour of https://github.com/tekdi/eg-backend/pull/645

**Pull request closed without merging**